src/routing_table/mod.rs

```diff
@@ -259,8 +260,8 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         }
     }
 
-    pub fn our_name(&self) -> &T {
-        &self.our_name
+    pub fn our_name(&self) -> T {
+        self.our_name
```

Comment:
Outside the tests it's a `[u8; 32]`, which is larger than a reference. Not sure how the overhead of dereferencing itself compares to that, though, but I'd stick to the rule of thumb of using references for anything but primitive types.